streams/src/main/java/org/apache/kafka/streams/processor/internals/CorruptedRecord.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor.internals;
+
+import java.util.Objects;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.header.Headers;
+
+/**
+ * This class represents a version of a {@link StampedRecord} that failed to deserialize. We need
+ * a special record type so that {@link StreamTask} can update consumed offsets. See KAFKA-6502
+ * for more details.
+ */
+public class CorruptedRecord extends StampedRecord {
+
+    private final ConsumerRecord<byte[], byte[]> rawRecord;
+
+    CorruptedRecord(final ConsumerRecord<byte[], byte[]> rawRecord) {
+        super(null, ConsumerRecord.NO_TIMESTAMP);
```

Comment:
Why not pass `rawRecord` (but replace with `null`)? Seems we could avoid overwriting all the methods (or I don't understand why we need to overwrite them)?

Reply:
We can't pass `rawRecord` to the constructor because its type is `ConsumerRecord<byte[], byte[]>` while `StampedRecord` expects `ConsumerRecord<Object, Object>`; `byte[]` does not extend `Object`.

Reply:
Ah. That's annoying... I am usually a bid advocate to use generic for type safety, but for this case, I am wondering if we should sacrifice it to avoid boiler plate code?

Reply:
Fixed in 19225a4.